### ISSUE TITLE
Warnings in the user guide

### ISSUE THREAD

**Issue by glen-anderson** [glen-anderson/msdk#3]
Lots of warnings that need to be addressed in the UG now that it's merged into main.

```
Documentation/user-guide/board-support-pkgs.md:3:64 MD051/link-fragments Link fragments should be valid [Context: "[supported parts](#supported-parts)"]
Documentation/user-guide/board-support-pkgs.md:14:20 MD051/link-fragments Link fragments should be valid [Context: "[How to Set the BSP (VS Code)](#how-to-set-the-bsp-vs-code)"]
Documentation/user-guide/board-support-pkgs.md:15:20 MD051/link-fragments Link fragments should be valid [Context: "[How to Set the BSP (Eclipse)](#how-to-set-the-bsp-eclipse)"]
Documentation/user-guide/board-support-pkgs.md:16:34 MD051/link-fragments Link fragments should be valid [Context: "[How to Set the BSP (Command-Line)](#how-to-set-the-bsp-command-line)"]
Documentation/user-guide/board-support-pkgs.md:24:53 MD051/link-fragments Link fragments should be valid [Context: "[build configuration variable](#build-configuration-variables)"]
Documentation/user-guide/board-support-pkgs.md:122:76 MD051/link-fragments Link fragments should be valid [Context: "[build configuration variable](#build-configuration-variables)"]
Documentation/user-guide/board-support-pkgs.md:126:147 MD051/link-fragments Link fragments should be valid [Context: "[build configuration variable](#build-configuration-variables)"]
Documentation/user-guide/board-support-pkgs.md:144:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/board-support-pkgs.md:163:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/board-support-pkgs.md:176:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/board-support-pkgs.md:192:98 MD051/link-fragments Link fragments should be valid [Context: "[build configuration variable](#build-configuration-variables)"]
Documentation/user-guide/board-support-pkgs.md:196:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/board-support-pkgs.md:203:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/build-system.md:90:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
Documentation/user-guide/build-system.md:104:186 MD051/link-fragments Link fragments should be valid [Context: "[Board Support Packages](#board-support-packages)"]
Documentation/user-guide/build-system.md:105:394 MD033/no-inline-html Inline HTML [Element: br]
Documentation/user-guide/build-system.md:105:402 MD051/link-fragments Link fragments should be valid [Context: "[BSP Search Directory](#bsp-search-directory)"]
Documentation/user-guide/build-system.md:117:231 MD051/link-fragments Link fragments should be valid [Context: "[project_name](#project_name)"]
Documentation/user-guide/build-system.md:118:210 MD033/no-inline-html Inline HTML [Element: br]
Documentation/user-guide/build-system.md:118:354 MD033/no-inline-html Inline HTML [Element: br]
Documentation/user-guide/build-system.md:118:425 MD033/no-inline-html Inline HTML [Element: br]
Documentation/user-guide/build-system.md:118:510 MD033/no-inline-html Inline HTML [Element: br]
Documentation/user-guide/build-system.md:130:236 MD055/table-pipe-style Table pipe style [Expected: leading_and_trailing; Actual: leading_only; Missing trailing pipe]
Documentation/user-guide/build-system.md:158:313 MD055/table-pipe-style Table pipe style [Expected: leading_and_trailing; Actual: leading_only; Missing trailing pipe]
Documentation/user-guide/build-system.md:164:51 MD051/link-fragments Link fragments should be valid [Context: "[available libraries](#libraries)"]
Documentation/user-guide/build-system.md:164:207 MD051/link-fragments Link fragments should be valid [Context: "[libraries](#libraries)"]
Documentation/user-guide/build-system.md:189:230 MD051/link-fragments Link fragments should be valid [Context: "[The SPI V2 Developer Note](#spi-v2-library)"]
Documentation/user-guide/build-system.md:209:198 MD055/table-pipe-style Table pipe style [Expected: leading_and_trailing; Actual: leading_only; Missing trailing pipe]
Documentation/user-guide/command-line-dev.md:5:28 MD051/link-fragments Link fragments should be valid [Context: "["Getting Started with Command-Line Development"](#getting-started-with-command-line-development)"]
Documentation/user-guide/command-line-dev.md:9:51 MD051/link-fragments Link fragments should be valid [Context: "[Build Configuration Variable](#build-configuration-variables)"]
Documentation/user-guide/command-line-dev.md:48:134 MD051/link-fragments Link fragments should be valid [Context: "[peripheral driver](#peripheral-driver-api)"]
Documentation/user-guide/command-line-dev.md:165:48 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/command-line-dev.md:165:55 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/debuggers.md:7:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/debuggers.md:8:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/debuggers.md:9:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/debuggers.md:69:40 MD051/link-fragments Link fragments should be valid [Context: "[build task](#build-tasks)"]
Documentation/user-guide/developer-notes.md:25:103 MD051/link-fragments Link fragments should be valid [Context: "[build configuration variable](#build-configuration-variables)"]
Documentation/user-guide/eclipse.md:3:41 MD051/link-fragments Link fragments should be valid [Context: "["Getting Started with Eclipse"](#getting-started-with-eclipse)"]
Documentation/user-guide/eclipse.md:32:30 MD051/link-fragments Link fragments should be valid [Context: "[_Board Support Package (BSP)_](#board-support-packages)"]
Documentation/user-guide/eclipse.md:67:25 MD051/link-fragments Link fragments should be valid [Context: "[Build Configuration Variable](#build-tables)"]
Documentation/user-guide/eclipse.md:69:9 MD051/link-fragments Link fragments should be valid [Context: "[Board Support Packages](#board-support-packages)"]
Documentation/user-guide/examples.md:3:87 MD051/link-fragments Link fragments should be valid [Context: "[Peripheral APIs](#peripheral-driver-api)"]
Documentation/user-guide/examples.md:15:1 MD020/no-missing-space-closed-atx No space inside hashes on closed atx style heading [Context: "##__EXAMPLES_LIST__##"]
Documentation/user-guide/examples.md:15:21 MD047/single-trailing-newline Files should end with a single newline character
Documentation/user-guide/examples.md:15:3 MD050/strong-style Strong style [Expected: asterisk; Actual: underscore]
Documentation/user-guide/examples.md:15:18 MD050/strong-style Strong style [Expected: asterisk; Actual: underscore]
Documentation/user-guide/getting-started.md:15:212 MD051/link-fragments Link fragments should be valid [Context: "[supported parts](#supported-parts)"]
Documentation/user-guide/getting-started.md:35:384 MD051/link-fragments Link fragments should be valid [Context: "["Supported Development Environments"](#supported-development-environments)"]
Documentation/user-guide/getting-started.md:55:37 MD051/link-fragments Link fragments should be valid [Context: "[Visual Studio Code](#visual-studio-code)"]
Documentation/user-guide/getting-started.md:59:53 MD051/link-fragments Link fragments should be valid [Context: "[installation](#installation)"]
Documentation/user-guide/getting-started.md:121:13 MD051/link-fragments Link fragments should be valid [Context: "[Board Support Packages](#board-support-packages)"]
Documentation/user-guide/getting-started.md:164:13 MD051/link-fragments Link fragments should be valid [Context: "[Visual Studio Code](#visual-studio-code)"]
Documentation/user-guide/getting-started.md:172:216 MD051/link-fragments Link fragments should be valid [Context: "[Maintenance Tool](#maintenance)"]
Documentation/user-guide/getting-started.md:172:111 MD051/link-fragments Link fragments should be valid [Context: "[MSDK installation](#installation)"]
Documentation/user-guide/getting-started.md:174:140 MD051/link-fragments Link fragments should be valid [Context: "[Eclipse](#eclipse)"]
Documentation/user-guide/getting-started.md:195:30 MD051/link-fragments Link fragments should be valid [Context: "[_Board Support Package (BSP)_](#board-support-packages)"]
Documentation/user-guide/getting-started.md:207:1 MD030/list-marker-space Spaces after list markers [Expected: 1; Actual: 2]
Documentation/user-guide/getting-started.md:224:13 MD051/link-fragments Link fragments should be valid [Context: "[Eclipse](#eclipse)"]
Documentation/user-guide/getting-started.md:232:42 MD051/link-fragments Link fragments should be valid [Context: "[Command-Line Development](#command-line-development)"]
Documentation/user-guide/getting-started.md:278:167 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/getting-started.md:278:181 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/getting-started.md:352:379 MD051/link-fragments Link fragments should be valid [Context: "[Flashing on the Command-Line](#flashing-on-the-command-line)"]
Documentation/user-guide/getting-started.md:390:13 MD051/link-fragments Link fragments should be valid [Context: "[Command-Line Development](#command-line-development)"]
Documentation/user-guide/index.md:21:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:27 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
Documentation/user-guide/index.md:30:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:38:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:43:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:51:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:57:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:65:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:71:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:77:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:87:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:93:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:99:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:107:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:113:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:120:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/index.md:130:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
Documentation/user-guide/installation.md:68 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
Documentation/user-guide/libraries.md:6:62 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/libraries.md:6:76 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/libraries.md:6:112 MD051/link-fragments Link fragments should be valid [Context: "[Build Variables for Toggling Libraries](#build-variables-for-toggling-libraries)"]
Documentation/user-guide/libraries.md:66:30 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:67:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:68:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:69:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:70:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:71:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:72:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:73:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:74:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:75:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:76:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:77:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:78:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:79:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:80:23 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:96:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:97:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:98:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:99:158 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:149:223 MD051/link-fragments Link fragments should be valid [Context: "[Board Support Packages](#board-support-packages)"]
Documentation/user-guide/libraries.md:153:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:154:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:155:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:156:554 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:162:36 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/libraries.md:162:43 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/libraries.md:164:5 MD051/link-fragments Link fragments should be valid [Context: "[Build Variables for Toggling Libraries](#build-variables-for-toggling-libraries)"]
Documentation/user-guide/libraries.md:176:29 MD051/link-fragments Link fragments should be valid [Context: "[build configuration variables](#build-configuration-variables)"]
Documentation/user-guide/libraries.md:178:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:179:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:180:380 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:181:251 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:182:256 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:196:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:197:150 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/libraries.md:198:413 MD055/table-pipe-style Table pipe style [Expected: leading_only; Actual: leading_and_trailing; Unexpected trailing pipe]
Documentation/user-guide/visual-studio-code.md:5:41 MD051/link-fragments Link fragments should be valid [Context: "["Getting Started with Visual Studio Code"](#getting-started-with-visual-studio-code)"]
Documentation/user-guide/visual-studio-code.md:48:190 MD051/link-fragments Link fragments should be valid [Context: "[Build Configuration Variable](#build-tables)"]
Documentation/user-guide/visual-studio-code.md:50:9 MD051/link-fragments Link fragments should be valid [Context: "[Board Support Packages](#board-support-packages)"]
Documentation/user-guide/visual-studio-code.md:151:16 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/visual-studio-code.md:151:34 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/visual-studio-code.md:155:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
Documentation/user-guide/visual-studio-code.md:156:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
Documentation/user-guide/visual-studio-code.md:157:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
Documentation/user-guide/visual-studio-code.md:158:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
Documentation/user-guide/visual-studio-code.md:160:57 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/visual-studio-code.md:160:68 MD049/emphasis-style Emphasis style [Expected: underscore; Actual: asterisk]
Documentation/user-guide/visual-studio-code.md:250:106 MD051/link-fragments Link fragments should be valid [Context: "[Getting Started with Visual Studio Code](#getting-started-with-visual-studio-code)"]
Documentation/user-guide/visual-studio-code.md:255:24 MD051/link-fragments Link fragments should be valid [Context: "[Build Configuration](#build-configuration-variables)"]
Documentation/user-guide/visual-studio-code.md:414:18 MD051/link-fragments Link fragments should be valid [Context: "[build system](#build-system)"]
```

**Comment by jkernion-adi** [comment]
Question re: examples.md errors on line 15 -- does the formatting of this line of code (##__EXAMPLES_LIST__##) affect the build of the examples list in any way? I can fix it to suit the linter but I don't want to break the include.

**Comment by glen-anderson** [comment]
Fixed in #5
